### ISSUE TITLE
How can i know when foreachInit has run?

### ISSUE THREAD

**Issue by boedlen** [ErikSchierboom/knockout-pre-rendered#45]
Hi Erik,

Is it possible to get a notification when the foreachInit method is done?

I have a table that has the `foreachInit` binding, and when it's applied my observable array is populated with items - assuming i rendered <tr>s from the server.

Now i'd like to show  a special message when my table doesn't have any row when rendered from the server - but i'm having a hard time knowing when my array is empty because foreachInit hasn'y run yet or because there were no rows in the table.

I tried using `beforeQueueFlush` and `afterQueueFlush` but they don't run when my table is empty.

Is there an event or callback that is triggered when no rows are found?

**Comment by ErikSchierboom** [comment]
Would you be content with a callback that is called before anything is rendered to the screen (which is more problematic due to asynchronicity), but after the data that is to rendered has been gathered? In other words, this callback would be called just before the changes are added to the queue to be rendered later.

**Comment by boedlen** [comment]
In my specific scenario, i'm interested in knowing when `foreachInit` has run, and how many items it found.

Zooming out and looking at a broader use case, couldn't `beforeQueueFlush` be called even on empty tables? 
Without having dug deeper into the source, it seems to do the right thing at the right time when my table is not empty.

Alternately a different callback that is called at the same time, but for empty tables aswell.

**Comment by ErikSchierboom** [comment]
`beforeQueueFlush` is called whenever the queue is actually processed, which is done asynchronously whenever changes are made. However, in your case there are no changes, so therefore `beforeQueueFlush` is never called. 

What I've done is added a new callback, named `dataChanged`. This callback is also called upon initialization, where the input array may be empty. Note that this callback changes from `beforeQueue` in that it only receives the changes being made, whereas `beforeQueue` always receives the complete queue of changes.

You can test this feature in the [data-change-callback](https://github.com/ErikSchierboom/knockout-pre-rendered/tree/data-change-callback) branch. Would you please let me know if it solves your use case? If so, I'll release a new version.

**Comment by boedlen** [comment]
Hi Erik,
Sorry for the late response.
I've finally had a chance to look at the added `dataChanged` callback, and it does exactly what i need :+1: 

A small note about the update to the docs:

```
 this.dataChanged = function(changes) {
    console.log(changes.length + ' items have been changed');
  }
```
Since the argument `changes` has this signature: 
```
{
    added: Array[0], 
    existing: Array[0], 
    deleted: Array[0]
}
```
Maybe the sample should be something like this instead
```
 this.dataChanged = function(changes) {
    console.log(changes.added.length+ ' items have been added');
    console.log(changes.existing.length + ' items existed');
    console.log(changes.deleted.length + ' items were deleted');
  }
```

**Comment by ErikSchierboom** [comment]
Great suggestion. I've updated the README and have just published the new version (0.8.0) that includes this change.
